core/src/main/java/com/microsoft/applicationinsights/telemetry/BaseTelemetry.java

```diff
@@ -160,7 +160,7 @@ public void serialize(JsonTelemetryDataSerializer writer) throws IOException {
         tmp.setBaseData(getData());
         tmp.setBaseType(this.getBaseTypeName());
         envelope.setData(tmp);
-        envelope.setTime(LocalStringsUtils.getDateFormatter().format(getTimestamp()));
+        envelope.setTime(getTimestamp() != null ? LocalStringsUtils.getDateFormatter().format(getTimestamp() : "null"));
```

Comment:
```suggestion
        if (getTimestamp() != null) envelope.setTime(LocalStringsUtils.getDateFormatter().format(getTimestamp());
```
Then https://github.com/microsoft/ApplicationInsights-Java/blob/e29b3d9118ca1ab59ff54ddedb6c53e35c278095/core/src/main/java/com/microsoft/applicationinsights/internal/schemav2/Envelope.java#L219-L231
and https://github.com/microsoft/ApplicationInsights-Java/blob/e29b3d9118ca1ab59ff54ddedb6c53e35c278095/core/src/main/java/com/microsoft/applicationinsights/telemetry/JsonTelemetryDataSerializer.java#L210-L216
take care of a default value.